### ISSUE TITLE
Train vs Test Accuracy

### ISSUE THREAD

**Issue by xttam** [deeplearning4j/deeplearning4j#3782]
On the UI it would be really useful to have a chart of the accuracy for both the training and testing dataset, instead of the current "Model Score vs. Iteration" chart.

This new chart should be similar to [1], which I think can be achieved by providing a testing dataset before starting the training process.

```java
model.evaluate(testing);
for (int i=0; i<epochs; i++) {
    model.fit(training);
}
```

In fact, this chunk of code above could be encapsulated so the model becomes aware of the number of epochs. This way all charts on the UI could be in units of epochs (x-axis) instead of iterations (as they currently are). As mentioned in [2], "It is preferable to track epochs rather than iterations since the number of iterations depends on the arbitrary setting of batch size."

```java
NeuralNetConfiguration.Builder()
    ...
    .epochs(epochs)
    ...
    .build()

model.fit(training, testing);
```

Matt

[1] https://cs231n.github.io/assets/nn3/accuracies.jpeg
[2] https://cs231n.github.io/neural-networks-3/#baby

**Comment by tim-bitreactive** [comment]
@raver119 Could you please refer to the issue this one duplicates?
